### PR TITLE
[1848] add boe stock row and initial price

### PR DESCRIPTION
**Pull request by bentziaxl**
added bank of england stock row, and a custom method to find boe share price.
added some cosmetic changes such as game trigger text.
fixed a minor bug in setup where the boe owner and starting cash was defined inside the share selling for loop.
fixed boe double share

### CODE REVIEW

**Comment by benjaminxscott** [comment]
typo "of more loans" 
https://github.com/tobymao/18xx/pull/8061/files#diff-f9b14b22c95c8c97908b4b327ff152658d3d149bdf56462e59d5b65812680356R117

**Comment by bentziaxl** [comment]
fixed

**Comment by bentziaxl** [comment]
ignore latest commit and revert, by accident

File: lib/engine/game/g_1848/game.rb [hunk 1]
```diff
@@ -487,6 +528,13 @@ def after_buy_company(player, company, _price)
             end
           end
         end
+
+        def lookup_boe_price(p)
+          @stock_market.market[BOE_ROW].size.times do |i|
```
Comment (crericha) [inline lib/engine/game/g_1848/game.rb:533]: .each instead of .size.times

Comment (bentziaxl) [reply lib/engine/game/g_1848/game.rb:533]: i copied this method from somewhere else, using each is way easier, i wonder why it's not used everywhere for price lookups.

File: lib/engine/game/g_1848/game.rb [hunk 2]
```diff
@@ -530,9 +530,8 @@ def after_buy_company(player, company, _price)
         end
 
         def lookup_boe_price(p)
-          @stock_market.market[BOE_ROW].size.times do |i|
-            next unless @stock_market.share_price(BOE_ROW, i)
-            return @stock_market.share_price(BOE_ROW, i) if @stock_market.share_price(BOE_ROW, i).price == p
+          @stock_market.market[BOE_ROW].each do |i|
```
Comment (crericha) [inline lib/engine/game/g_1848/game.rb:533]: It's best practice to be more descriptive in the block args to give context of what is being iterated. For example, here I'd used sp for share_price.

Comment (crericha) [reply lib/engine/game/g_1848/game.rb:533]: Let me know if you want to make that change before I merge.

**Review by crericha — APPROVED** [review]

File: lib/engine/game/g_1848/game.rb (hunk 1 above)
Comment (crericha) [reply lib/engine/game/g_1848/game.rb]: I'm not sure how the .times snuck in. Usually Toby (the primary reviewer) is good at catching those.

File: lib/engine/game/g_1848/game.rb (hunk 2 above)
Comment (bentziaxl) [reply lib/engine/game/g_1848/game.rb]: done. can merge

Comment (crericha) [reply lib/engine/game/g_1848/game.rb:533]: I was referring to the block variable - i, which represents an array of share prices. It makes it easier to maintain when the block variable gives context as to what it is. the method param was fine -- though price would have been more descriptive there.

Comment (bentziaxl) [reply lib/engine/game/g_1848/game.rb]: done

**Review by crericha — APPROVED** [review]

**Pull request merged**